base_layer/common_types/src/tari_address/mod.rs

```diff
@@ -20,74 +20,106 @@
 // WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 // USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+mod dual_address;
+mod single_address;
+
 use std::{
-    convert::TryFrom,
     fmt::{Display, Error, Formatter},
     str::FromStr,
 };
 
+use bitflags::bitflags;
 use serde::{Deserialize, Serialize};
 use tari_common::configuration::Network;
 use tari_crypto::tari_utilities::ByteArray;
 use tari_utilities::hex::{from_hex, Hex};
 use thiserror::Error;
 
 use crate::{
-    dammsum::{compute_checksum, validate_checksum},
-    emoji::{EMOJI, REVERSE_EMOJI},
+    emoji::EMOJI,
+    tari_address::{dual_address::DualAddress, single_address::SingleAddress},
     types::PublicKey,
 };
 
-const INTERNAL_SIZE: usize = 33; // number of bytes used for the internal representation
+const INTERNAL_DUAL_SIZE: usize = 67; // number of bytes used for the internal representation
+const INTERNAL_SINGLE_SIZE: usize = 35; // number of bytes used for the internal representation
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct TariAddressFeatures(u8);
```

Comment:
How are these different from the NodeFeatures, and why did we decide to make them an independent feature set?

Reply:
yes they are, node features are coms feature sets. 
This is all solely aimed at the wallet